### PR TITLE
Upgradeable ERC-20 Tokens

### PR DESCRIPTION
**Pull request by kacperzuk-neti**

### CODE REVIEW

File: eth-bridge/contracts/src/Bridge.sol [hunk 1]
```diff
@@ -6,31 +6,8 @@ import {UUPSUpgradeable} from "openzeppelin-contracts-upgradeable/contracts/prox
 import {AccessControlUpgradeable} from
     "openzeppelin-contracts-upgradeable/contracts/access/AccessControlUpgradeable.sol";
 import {WrappedToken} from "./WrappedToken.sol";
-
-/// Struct for representing Substrate -> ETH transfer
-struct IncomingReceiptStruct {
```
Comment (kacperzuk-neti) [inline eth-bridge/contracts/src/Bridge.sol:11]: Extracted to `BridgeTypes.sol`, as new solhint version doesn't like multiple contracts in one file.

File: eth-bridge/contracts/src/Bridge.sol [hunk 2]
```diff
@@ -59,37 +36,6 @@ error AlreadyVoted();
 /// Transferred amount is less than configured minimum
 error TooSmallAmount();
 
-/// @title Interface with events emitted by the bridge
```
Comment (kacperzuk-neti) [inline eth-bridge/contracts/src/Bridge.sol:62]: Extracted to `BridgeEvents.sol`, as new solhint version doesn't like multiple contracts in one file.

File: eth-bridge/contracts/lib/forge-std
Comment (kacperzuk-neti) [inline eth-bridge/contracts/lib/forge-std:1]: Update of dependencies was required to get `ERC20PermitUpgradeable` from openzeppelin

File: .github/workflows/test-solidity.yml [hunk 3]
```diff
@@ -53,7 +53,7 @@ jobs:
         working-directory: ./eth-bridge/contracts/
         run: |
           forge --version
-          forge build --sizes
+          forge build --build-info --sizes
```
Comment (kacperzuk-neti) [inline .github/workflows/test-solidity.yml:56]: New version of slither requires this.

File: .github/workflows/test-solidity.yml [hunk 4]
```diff
@@ -73,4 +73,4 @@ jobs:
         with:
           target: ./eth-bridge/contracts/
           slither-config: ./eth-bridge/contracts/slither.config.json
-          ignore-compilation: true
+          ignore-compile: true
```
Comment (kacperzuk-neti) [inline .github/workflows/test-solidity.yml:76]: New version of slither requires this.

**Review by DorianSternVukotic — APPROVED** [review]

**Pull request merged**